assets/src/stories-editor/components/higher-order/with-keyboard-navigation-handler.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * WordPress dependencies
+ */
+import { withDispatch } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { UP, DOWN, RIGHT, LEFT } from '@wordpress/keycodes';
+import { KeyboardShortcuts } from '@wordpress/components';
+/**
+ * Internal dependencies
+ */
+import { ALLOWED_CHILD_BLOCKS, ALLOWED_MOVABLE_BLOCKS } from '../../constants';
+
+const applyWithDispatch = withDispatch( ( dispatch, props, { select } ) => {
+	const {	isReordering } = select( 'amp/story' );
+	const { getSelectedBlock } = select( 'core/block-editor' );
+	const { updateBlockAttributes, removeBlock } = dispatch( 'core/block-editor' );
+	const selectedBlock = getSelectedBlock();
+
+	const onMoveBlock = ( event ) => {
+		const { keyCode, target } = event;
+		const { classList } = target;
+
+		if ( ! selectedBlock ) {
+			return;
+		}
+
+		if ( classList.contains( 'editor-rich-text__editable' ) && classList.contains( 'is-selected' ) ) {
+			return;
+		}
+
+		let top = 0;
+		let left = 0;
+		switch ( keyCode ) {
```

Comment:
Needs some testing, but it would be nice if one could move in bigger steps when holding down an additional key, e.g. ctrl, alt, or cmd. If I am not mistaken this kind of behavior is common in other applications when moving elements with the keyboard